terminal2048/lib/terminal2048/player.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Terminal2048.Player do
+
+  def play() do
+    play(Terminal2048.Board.new_game)
+  end
+
+  defp play(game) do
+    get_next_move({game, game.game_state})
+  end
+
+  defp get_next_move({game, :lost}) do
+    draw_current_board(game)
+    IO.puts("\nSorry, you lose.")
+  end
+
+  defp get_next_move({game, :won}) do
+    draw_current_board(game)
+    IO.puts("\nYou win!")
+  end
+
+  defp get_next_move({game, _game_state}) do
+    draw_current_board(game)
+    move = "\nMake your move: "
+      |> IO.gets
+      |> String.downcase
+      |> String.trim
+      |> get_move
+
+    Terminal2048.Board.make_move(game, move)
+    |> get_next_move
+  end
+
+  def draw_current_board(game) do
+    clear_screen()
+    IO.puts(Terminal2048.Board.draw_current_board(game))
+  end
+
+  defp clear_screen(), do: IO.write("\e[H\e[2J")
+
+  defp get_move("w"), do: :up
+  defp get_move("s"), do: :down
+  defp get_move("a"), do: :left
+  defp get_move("d"), do: :right
+  defp get_move(_) do
+    "\nInvalid move. Try another move: "
```

Comment:
I don't like the duplication here.

If `display_board` was moved outside `get_next_move`, then you could simply call it from here. 

This is a good example of how a "code smell" (the duplication) is a pointer to a better design.

# -2